### ISSUE TITLE
Unable to configure vector intrinsic when using clang

### ISSUE THREAD

**Issue by oroppas** [flame/libflame#16]
configure doesn't know how to set sse flag for `clang`

```
checking whether user requested use of vector intrinsics... yes
checking user-requested vector intrinsic type... sse
checking for (guessing) SSE flags for /home/ryutas/local/bin/clang... unknown
configure: error: configure doesn't know what flag to give /home/ryutas/local/bin/clang in order to enable SSE. Please submit a bug report to the FLAME developers.
```

**Comment by jeffhammond** [comment]
Can you provide the full configure invocation, Clang version and platform details?

**Comment by oroppas** [comment]
../libflame/configure \
  --prefix=$HOME/local \
  --enable-lapack2flame \
  --enable-vector-intrinsics=sse

clang --version
clang version 6.0.0 (tags/RELEASE_600/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix

Linux 3.10.0-514.6.1.el7.x86_64

**Comment by oroppas** [comment]
../libflame/configure \
  --prefix=$HOME/local \
  --enable-lapack2flame \
  --enable-vector-intrinsics=sse

clang --version
clang version 6.0.0 (tags/RELEASE_600/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix

Linux 3.10.0-514.6.1.el7.x86_64

**Comment by jeffhammond** [comment]
I cannot reproduce this with Ubuntu 18.04 (ubuntu-18-04 4.15.0-48-generic) and Clang 6.0.0.

**Comment by jeffhammond** [comment]
Can you verify that the `-msse3` flag works with your compiler with a trivial program?

**Comment by oroppas** [comment]
Actually, how I configure was not exactly like I posted since I set something

```
CC=/usr/local/bin/clang \
../libflame/configure \
...
```

and setting `CC` with absolute path confused the `configure`.

Thank you for verifying on ubuntu.

**Comment by jeffhammond** [comment]
I guess it’s not a high priority but we should fix configure to let you do this. I’m sure there’s some easy shell magic to strip the path so that the compiler name can be processed appropriately.

**Comment by fgvanzee** [comment]
> I guess it’s not a high priority but we should fix configure to let you do this. I’m sure there’s some easy shell magic to strip the path so that the compiler name can be processed appropriately.

I don't even think that's the issue. I tried:
```
CC=/usr/bin/clang ./configure
```
and libflame seemed to proceed with compilation just fine. Running `make V=1`, I was able to confirm that `/usr/bin/clang` was indeed being run, as expected.

The problem is likely the fact that @oroppas was trying to build out-of-tree, which I did not explicitly design libflame's build system to support. (It may *almost* work by dumb luck, but I suspect the current use of the `version` file breaks out-of-tree building.)
